clientLibrary/src/main/scala/za/co/absa/loginclient/tokenRetrieval/service/RetrieveToken.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginclient.tokenRetrieval.service
+
+import com.google.gson.{JsonObject, JsonParser}
+import org.slf4j.{Logger, LoggerFactory}
+import org.springframework.http.{HttpEntity, HttpHeaders, HttpMethod, MediaType, ResponseEntity}
+import org.springframework.web.client.RestTemplate
+import za.co.absa.loginclient.tokenRetrieval.model.{AccessToken, RefreshToken}
+
+import java.net.URLEncoder
+import java.util.Collections
+
+/**
+ * This class is used to retrieve tokens from the login service.
+ * Refresh and Access Keys require authorization. Basic Auth is used for the initial retrieval.
+ * Refresh token from initial retrieval is used to refresh the access token.
+ * @param host The host of the login service.
+ */
+
+case class RetrieveToken(host: String) {
```

Comment:
Just some naming convention: Usually, a service class is nominalized, often with "Service" at the end. E.g. `TokenRetrievalService`. Same for the `RetrievePublicKey` service